pkg/kubefedctl/orphaning/orphaning.go

```diff
@@ -0,0 +1,130 @@
+limitations under the License.
+*/
+
+package orphaning
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/klog"
+
+	ctlutil "sigs.k8s.io/kubefed/pkg/controller/util"
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/enable"
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/options"
+	"sigs.k8s.io/kubefed/pkg/kubefedctl/util"
+)
+
+type orphanResource struct {
+	options.GlobalSubcommandOptions
+	typeName          string
+	resourceName      string
+	resourceNamespace string
+}
+
+// Bind adds the join specific arguments to the flagset passed in as an argument.
+func (o *orphanResource) Bind(flags *pflag.FlagSet) error {
+	flags.StringVarP(&o.resourceNamespace, "namespace", "n", "default", "If present, the namespace scope for this CLI request")
+	err := flags.MarkHidden("kubefed-namespace")
+	if err != nil {
+		return err
+	}
+	err = flags.MarkHidden("dry-run")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// NewCmdOrphaning the head of orphaning-deletion sub commands
+func NewCmdOrphaning(cmdOut io.Writer, config util.FedConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "orphaning-deletion",
+		Short: "Manage orphaning delete policy",
+		Long:  "Manage orphaning delete policy",
+		Run: func(cmd *cobra.Command, args []string) {
+			err := cmd.Help()
+			if err != nil {
+				klog.Fatalf("Error: %v", err)
+			}
+		},
+	}
+	cmd.AddCommand(newCmdEnableOrphaning(cmdOut, config))
+	cmd.AddCommand(newCmdDisableOrphaning(cmdOut, config))
+	cmd.AddCommand(newCmdStatusOrphaning(cmdOut, config))
+
+	return cmd
+}
+
+//  Complete ensures that options are valid and marshals them if necessary.
+func (o *orphanResource) Complete(args []string) error {
+	if len(args) == 0 {
+		return errors.New("resource type is required")
+	}
+
+	o.typeName = args[0]
+
+	if len(args) == 1 {
+		return errors.New("resource name is required")
+	}
+	o.resourceName = args[1]
+
+	return nil
+}
+
+// Returns a Federated Resources Interface
+func (o *orphanResource) GetResourceClient(config util.FedConfig, cmdOut io.Writer) (dynamic.ResourceInterface, error) {
+	hostConfig, err := config.HostConfig(o.HostClusterContext, o.Kubeconfig)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Unable to load configuration for cluster context %q in kubeconfig %q.`",
+			o.HostClusterContext, o.Kubeconfig)
+	}
+	// Lookup kubernetes API availability
+	apiResource, err := enable.LookupAPIResource(hostConfig, o.typeName, "")
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to find targeted %s type", o.typeName)
+	}
+	klog.V(2).Infof("API Resource for %s.%s/%s found", apiResource.Name, apiResource.Group, apiResource.Version)
+	if !util.IsFederatedAPIResource(apiResource.Kind, apiResource.Group) {
+		fmt.Fprintf(cmdOut, "Warning: looks like the resource %s.%s/%s is not a Federated object\n",
```

Comment:
`s/looks like the resource//`
`s/is not a Federated object/does not appear to be a federated resource/`

As to how to compose name and group to avoid a trailing `.` for core groups, please use the [GroupQualifiedName function](https://github.com/marun/kubefed/blob/fix-namespaced-prop-status/pkg/apis/core/typeconfig/util.go#L34).

Reply:
I was not aware about the `GroupQualifiedName` function, thanks.
I changed the warning message according to your suggestion, however, as I understood the function `IsFederatedAPIResource` cannot 100% guarantee that the checked resource is not a Federated resource. Therefore, I'd suggest a  weaker statement. 
What do you think about `might not be a federated resource`

Reply:
I think `does not appear to be` is equivalent to `might not be` (i.e. `this does not appear to be correct` vs `this is not correct`).

Reply:
Done